src/nvim/undo.c

```diff
@@ -1303,43 +1315,42 @@ void u_read_undo(char_u *name, char_u *hash, char_u *orig_name)
   }
 
   /* Read undo data for "U" command. */
-  str_len = get4c(fp);
+  str_len = undo_read_4c(&bi);
   if (str_len < 0)
     goto error;
   if (str_len > 0)
-    line_ptr = READ_STRING(fp, str_len);
-  line_lnum = (linenr_T)get4c(fp);
-  line_colnr = (colnr_T)get4c(fp);
+    line_ptr = undo_read_string(&bi, (size_t)str_len);
```

Comment:
:+1: I cleaned up the whole function now, since it is touched in so many places already.